src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/DeliverySessionService.kt

```diff
@@ -35,8 +35,12 @@ class DeliverySessionService(
   val appointmentRepository: AppointmentRepository,
 ) {
   fun createUnscheduledSessionsForActionPlan(approvedActionPlan: ActionPlan) {
+    val previouslyApprovedSessions = approvedActionPlan.referral.approvedActionPlan?. let {
+      deliverySessionRepository.findAllByActionPlanId(it.id).count()
+    } ?: 0
+
     val numberOfSessions = approvedActionPlan.numberOfSessions!!
```

Comment:
what happens when numberOfSession <= previouslyApprovedSessions?

Reply:
I was told that could never happen so there was no need to check for it. Is this still true?

Reply:
Imo we should actively reject incorrect state -- it's a business rule that new approved sessions must be >= previous approved sessions, so if that's violated, we need to

1. not reduce the number of sessions
2. alert us